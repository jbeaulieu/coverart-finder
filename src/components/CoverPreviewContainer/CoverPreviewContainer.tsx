import Download from "@mui/icons-material/Download";
import { Button } from "@mui/material";
import './styles.css';

export type Props = {
  previewSrc?: string;
  downloadSrc?: string;
};

const CoverPreviewContainer = (props: Props) => {

  const { previewSrc, downloadSrc } = props;

  const downloadImage = async (url: string) => {
    await fetch(url)
      .then((response) => response.blob())
      .then((blob) => {
        // Create a link element, hide it, direct it towards the blob, and then 'click' it programatically
        const a = document.createElement("a");
        a.style = "display: none";
        document.body.appendChild(a);
        // Create a DOMString representing the blob and point the link element towards it
        const url = window.URL.createObjectURL(blob);
        a.href = url;
        a.download = 'cover.jpg';
        // Click the link to trigger the download
        a.click();
        // Release the reference to the file by revoking the Object URL
        window.URL.revokeObjectURL(url);
        // Remove the invisible link element from the DOM
        a.remove();
      })
      .catch(error => {
        console.log(error);
      });
  };

  return (
    <div className="preview-container">
      <img src={previewSrc} alt="cover art preview" width={300} height={300} style={{objectFit: 'cover'}} />
      <Button variant="contained" endIcon={<Download />} onClick={() => downloadImage(downloadSrc!)}>Download</Button>
    </div>
  );
}

export default CoverPreviewContainer;
