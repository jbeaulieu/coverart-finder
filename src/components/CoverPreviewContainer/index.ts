import CoverPreviewContainer from './CoverPreviewContainer';

export default CoverPreviewContainer;
