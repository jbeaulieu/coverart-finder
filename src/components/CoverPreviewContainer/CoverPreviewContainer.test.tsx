import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import CoverPreviewContainer from './CoverPreviewContainer';
import type { Props } from './CoverPreviewContainer';

const defaultProps: Props = {
  previewSrc: undefined,
  downloadSrc: undefined,
};

const renderCoverPreviewContainer = (props = defaultProps) => {
  return render(<CoverPreviewContainer {...props} />);
};

describe('Component CoverPreviewContainer', () => {

  it('renders without crashing', () => {
    const { asFragment } = renderCoverPreviewContainer();

    expect(asFragment()).toMatchSnapshot();
  });
});
