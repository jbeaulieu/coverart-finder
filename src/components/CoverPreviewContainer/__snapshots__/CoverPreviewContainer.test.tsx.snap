// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component CoverPreviewContainer > renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="preview"
    style="display: flex; flex-direction: column; align-items: center;"
  >
    <img
      alt="cover art preview"
      height="300"
      style="object-fit: cover;"
      width="300"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Download
    </button>
  </div>
</DocumentFragment>
`;
